tests/test_models.py

```diff
@@ -83,6 +83,81 @@ def test_model_singlepsr_noise_dip_cusp(nodmx_psrs,caplog):
     x0 = {pname:p.sample() for pname,p in zip(m.param_names, m.params)}
     m.get_lnlikelihood(x0)
 
+def test_model_singlepsr_noise_dm_nondiag(nodmx_psrs,caplog):
+    # caplog.set_level(logging.CRITICAL)
+    mn=models.model_singlepsr_noise(nodmx_psrs[1], dm_var=True,
+                                    dm_type='gp', dmgp_kernel ='nondiag',
+                                    dm_nondiag_kernel ='dmx_like')
+    assert hasattr(mn,'get_lnlikelihood')
+    x0 = {pname:p.sample() for pname,p in zip(mn.param_names, mn.params)}
+    mn.get_lnlikelihood(x0)
+
+    mn=models.model_singlepsr_noise(nodmx_psrs[1], dm_var=True,
+                                    dm_type='gp', dmgp_kernel ='nondiag',
+                                    dm_nondiag_kernel ='sq_exp')
+    assert hasattr(mn,'get_lnlikelihood')
+    x0 = {pname:p.sample() for pname,p in zip(mn.param_names, mn.params)}
+    mn.get_lnlikelihood(x0)
+    
+    mn=models.model_singlepsr_noise(nodmx_psrs[1], dm_var=True,
+                                    dm_type='gp', dmgp_kernel ='nondiag',
+                                    dm_nondiag_kernel ='sq_exp_rfband')
+    assert hasattr(mn,'get_lnlikelihood')
+    x0 = {pname:p.sample() for pname,p in zip(mn.param_names, mn.params)}
+    mn.get_lnlikelihood(x0)
+
+    mn=models.model_singlepsr_noise(nodmx_psrs[1], dm_var=True,
+                                    dm_type='gp', dmgp_kernel ='nondiag',
+                                    dm_nondiag_kernel ='periodic')
+    assert hasattr(mn,'get_lnlikelihood')
+    x0 = {pname:p.sample() for pname,p in zip(mn.param_names, mn.params)}
+    mn.get_lnlikelihood(x0)
+    
+    mn=models.model_singlepsr_noise(nodmx_psrs[1], dm_var=True,
+                                    dm_type='gp', dmgp_kernel ='nondiag',
+                                    dm_nondiag_kernel ='periodic_rfband')
+    assert hasattr(mn,'get_lnlikelihood')
+    x0 = {pname:p.sample() for pname,p in zip(mn.param_names, mn.params)}
+    mn.get_lnlikelihood(x0)
+
+    mn=models.model_singlepsr_noise(nodmx_psrs[1], dm_var=True,
+                                    dm_type='gp', dmgp_kernel ='nondiag',
+                                    dm_nondiag_kernel ='None')
```

Comment:
"None" isn't one of the options for `nondiag_kernel`, so the code will go to the end of the `if`-`elif` tree in [`dm_noise_block`](https://github.com/nanograv/enterprise_extensions/blob/f8db598d31840bad0d7845388b92e6afdb2da547/enterprise_extensions/blocks.py#L297) having never set `dm_prior` or `dm_basis`